Cabal/Distribution/PackageDescription/Parsec.hs

```diff
@@ -306,6 +307,16 @@ goSections specVer = traverse_ process
             commonStanzas <- use stateCommonStanzas
             name' <- parseUnqualComponentName pos args
             flib  <- lift $ parseCondTree' (foreignLibFieldGrammar name')  commonStanzas fields
+
+            let hasType ts = foreignLibType ts /= foreignLibType mempty
```

Comment:
> `foreignLibType mempty`

Can we write `mempty :: ForeignLibType` or `ForeignLibTypeUnknown` instead? Likewise for test suites and benchmarks.

Reply:
I'd like to refactor that soon after the branch is cut...

Reply:
Fine.